extensions/route_map/search.rs

```diff
@@ -0,0 +1,105 @@
+use pyo3::prelude::*;
+use std::borrow::Cow;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+use crate::util::normalize_path;
+use crate::{get_base_components, util, wrappers, HandlerGroup, Node, RouteMap};
+
+pub(crate) fn find_insert_handler_group<'a>(
+    root: &'a mut Node,
+    plain_routes: &'a mut HashMap<String, HandlerGroup>,
+    path: &str,
+    path_parameters: &PyAny,
+    is_static: bool,
+    handler_group: HandlerGroup,
+) -> PyResult<&'a mut HandlerGroup> {
```

Comment:
Yeah, originally the code inspected/added to the returned handler_group, but since I changed it to just provide a group to merge, the return value isn't needed anymore. Will remove.